web/packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -952,12 +960,13 @@ export default class Codec {
   }
 
   // | message type (29) | data_length uint32 | data []byte |
-  decodeRDPFastPathPDU(buffer: ArrayBuffer): RDPFastPathPDU {
+  decodeRDPFastPathPDU(buffer: ArrayBuffer): RdpFastPathPdu {
+    const dv = new DataView(buffer);
     let offset = 0;
     offset += byteLength; // eat message type
+    const dataLength = dv.getUint32(offset);
     offset += uint32Length; // eat data_length
-    const data = buffer.slice(offset);
-    return new RDPFastPathPDU(new Uint8Array(data));
+    return new Uint8Array(buffer, offset, dataLength);
```

Comment:
Does `new Uint8Array` allocate new memory and copy into it? If so, we've lost the advantage of zero copy from wasm memory, right?

Is there a way to slice into the existing `ArrayBuffer` without reallocating?

Reply:
Edit: looks like `Uint8Array`'s constructor _DOES NOT_ copy when provided an existing buffer, so this should be good!